meter/sma/listener.go

```diff
@@ -104,15 +104,47 @@ type Listener struct {
 }
 
 // New creates a Listener
-func New(log *util.Logger) (*Listener, error) {
+func New(log *util.Logger, network string) (*Listener, error) {
+	//Select local network interface
+	var iface *net.Interface = nil
+	if network != "" {
+		interfaces, err := net.Interfaces()
+		if err != nil {
+			return nil, fmt.Errorf("error resolving network interfaces: %w", err)
+		}
+
+		for i := range interfaces {
+			addresses, err := interfaces[i].Addrs()
+			if err != nil {
+				return nil, fmt.Errorf("error resolving IP addresses network interface %s: %w", interfaces[i].Name, err)
+			}
+
+			for n := range addresses {
+				if addresses[n].(*net.IPNet).Contains(net.ParseIP(network)) && iface == nil {
```

Comment:
Siehe https://stackoverflow.com/questions/23558425/how-do-i-get-the-local-ip-address-in-go#23558495. Dein cast scheint nicht sicher?